bin/regenerate_files.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+
+import json
+import os
+import shutil
+import subprocess
+import re
+
+import bazel_util
+
+def should_copy(src, dest):
+    if not os.path.exists(dest):
+        return True
+    p = subprocess.Popen(['diff', '-u', src, dest], stdout=subprocess.PIPE)
+    (stdout, _) = p.communicate()
+    if src.endswith('.pb.go'):
+        # there might be diffs for 'source:' lines and others.
+        has_diff = False
+        linecount = 0
+        for l in stdout.split('\n'):
+            linecount += 1
+            if not l:
+                continue
+            if linecount < 3:
+                # first two lines are headers, skipping
+                continue
+            if l.startswith('@@ '):
+                # header for a diff chunk
+                continue
+            if l.startswith(' '):
+                # part of non-changes
+                continue
+            if re.search(r'genfiles/.*\.proto\b', l):
+                # ignoring the proto file path -- the exact file path can be different,
+                # depending on the platform.
+                continue
+            if re.search(r'// \d+ bytes of a gzipped FileDescriptorProto', l):
+                # header comment for descriptor bytes data. It can be different if the file path
+                # is different.
+                continue
+            if re.match(r'^.\s*(0x[0-9a-f][0-9a-f],\s*)+$', l):
+                # file descriptor bytes data. It can be different if the file path is different.
+                continue
+            # Other changes would be meaningful changes.
+            has_diff = True
+            break
+    else:
+        has_diff = (stdout != '')
+    return has_diff
+
+def get_generated_files(WKSPC, genfiles):
+    lst = []
+    proto_libraries = 'kind("go_proto_library", "//...")'
+    proto_compiles = 'attr("langs", "go", kind("proto_compile", "//..."))'
+    genrules = 'attr("outs", ".go", kind("genrule", "//..."))'
+    bazel_util.bazel_build(bazel_util.bazel_query('+'.join([proto_libraries, proto_compiles, genrules])))
+
+    for proto in bazel_util.bazel_query('filter("\.proto$", labels("srcs", %s))' % proto_libraries):
```

Comment:
I suggest adding examples of input -> output for each case, in the very likely case, bazel breaks this again.